src/main/java/carpet/mixins/ServerGamePacketListenerImplMixin.java

```diff
@@ -24,6 +26,10 @@ private void onCustomCarpetPayload(ServerboundCustomPayloadPacket packet, Callba
         ResourceLocation channel = packet.getIdentifier();
         if (CarpetClient.CARPET_CHANNEL.equals(channel))
         {
+            // We should force onto the main thread here
+            // ServerNetworkHandler.handleData can possibly mutate data that isn't
+            // thread safe, and also allows for client commands to be executed
```

Comment:
```suggestion
            // thread safe, and also allows for client commands to be executed
            // [TODO] need to investigate if that is actually an issue since carpet doesn't fiddle with its packets.
```